File: core/const.py
import os

# LocalOrz
VERSION = '0.1 dev'
CORE_ERROR = 'Core Error'
JUDGE_CLIENT_PATH = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'judge_client')

# Problem Status
UNKNOWN = 'Unknown'
NORMAL = 'Normal'
COMPILATION_ERROR = 'Compilation Error'
NO_ADDITION_FILE = 'No Addition File'
PROBLEM_STATUS = (CORE_ERROR, UNKNOWN, NORMAL, COMPILATION_ERROR, NO_ADDITION_FILE)

# Testcase Status
ACCEPTED = 'Accepted'
WRONG_ANSWER = 'Wrong Answer'
PARTLY_ACCEPTED = 'Partly Accepted'
CANNOT_EXECUTE = 'Cannot Execute'
TIME_LIMIT_EXCEEDED = 'Time Limit Exceeded'
MEMORY_LIMIT_EXCEEDED = 'Memory Limit Exceeded'
RUNTIME_ERROR = 'Runtime Error'
PROGRAM_NO_OUTPUT = 'Program No Output'
NO_SOURCE_FILE = 'No Source File'
NO_STANDARD_INPUT = 'No Standard Input'
NO_STANDARD_OUTPUT = 'No Standard Output'
CHECKER_ERROR = 'Checker Error'
TESTCASE_STATUS = (ACCEPTED, WRONG_ANSWER, PARTLY_ACCEPTED, CANNOT_EXECUTE, TIME_LIMIT_EXCEEDED, MEMORY_LIMIT_EXCEEDED, RUNTIME_ERROR, PROGRAM_NO_OUTPUT, NO_SOURCE_FILE, NO_STANDARD_INPUT, NO_STANDARD_OUTPUT, CHECKER_ERROR, COMPILATION_ERROR, CORE_ERROR)

# Checker
NORMAL_JUDGE = 'Normal Judge'
FLOAT_JUDGE = 'Float Judge'
SPECIAL_JUDGE = 'Special Judge'
NORMAL_JUDGE_PATH = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'normal_judge')
FLOAT_JUDGE_PATH = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'float_judge')
CHECKER = [NORMAL_JUDGE, FLOAT_JUDGE, SPECIAL_JUDGE]

File: const.js
var path = require('path');

module.exports = {
    VERSION: '0.0.1',

    COMPARISON: {
        IGNORE: 'ignore',
        BYTE  : 'byte by byte',
        SPJ   : 'special judge'
    },

    RESULT: {
        UNKNOWN           : 'unknown',
        NORMAL            : 'normal',
        COMPILATION_ERROR : 'compilation error',
        NO_SOURCE_FILE    : 'no source file',
    },

    POINT: {
        AC             : 'accepted',
        WA             : 'wrong answer',
        RE             : 'runtime error',
        MLE            : 'memory limit exceeded',
        TLE            : 'time limit exceeded',
        PART_CORRECT   : 'partly correct',
        CANNOT_EXECUTE : 'cannot execute',
        SPJ_ERROR      : 'special judge error',
        NO_OUTPUT      : 'no output file',
        NO_STD_INPUT   : 'no standard input file',
        NO_STD_OUTPUT  : 'no standard output file',
        UNKNOWN        : 'unknown',
    },

    PERSON: {
        UNJUDGED : 'unjudged',
        JUDGED   : 'judged',
    },

    BIN: {
        TRACKER      : path.join(__dirname, 'bin', 'tracker'),
        NORMAL_JUDGE : path.join(__dirname, 'bin', 'normal_judge'),
    }
};
